src/lang/KclProvider.tsx

```diff
@@ -50,6 +53,29 @@ export function KclContextProvider({
     })
   }, [])
 
+  // Add format code to command palette.
+  useEffect(() => {
+    const commands: Command[] = [
+      {
+        name: 'format-code',
+        displayName: 'Format Code',
+        description: 'Nicely formats the KCL code in the editor.',
+        needsReview: false,
+        groupId: 'code',
+        icon: 'code',
+        onSubmit: (data) => {
+          kclManager.format()
+        },
+      },
+    ]
+    commandBarSend({
+      type: 'Add commands',
+      data: {
+        commands,
+      },
+    })
```

Comment:
I would add a teardown here so that whenever the KclProvider is removed from the vDOM the commands are definitely removed. There's a "Remove commands" event you can send in the returned function from this useEffect, similar to [what we do in SettingsAuthProvider](https://github.com/KittyCAD/modeling-app/blob/main/src/components/SettingsAuthProvider.tsx#L217-L221). KclProvider should theoretically never be unmounted since it wraps the root route though, so I'm realizing this is maybe a nit sorry I've typed too much to go back now :V

Reply:
No, it's not a nit. Plan for code changing and moving. Code as is creates a leak in the future that's hard to track down. I will definitely fix it.